pg_shard--1.0--1.1.sql

```diff
@@ -12,7 +12,16 @@ AS $sync_table_metadata_to_citus$
 		table_relation_id CONSTANT oid NOT NULL := table_name::regclass::oid;
 		dummy_shard_length CONSTANT bigint := 0;
 	BEGIN
-		-- copy shard placement metadata
+		-- grab lock for upsert
+		LOCK TABLE pg_dist_shard_placement IN EXCLUSIVE MODE;
+
+		-- perform update of shard health
+		UPDATE pg_dist_shard_placement
```

Comment:
I think we need to consider two things here:
* Shard placements must be identified not by shardid alone. We should use (shard_id + nodename + nodeport)
* Also, we need to consider only the shards that belong to table_relation_id.

So, I think this statement should look like as below:
```sql
-- perform update of shard health
UPDATE pg_dist_shard_placement
SET    shardstate = shard_placement.shard_state
FROM   pgs_distribution_metadata.shard_placement
WHERE  shardid = shard_placement.shard_id AND
       nodename = shard_placement.node_name AND
       nodeport = shard_placement.node_port AND
       shardid IN (SELECT id FROM   pgs_distribution_metadata.shard
		         WHERE  relation_id = table_relation_id);
```